lib/class-wp-annotation-caps.php

```diff
@@ -0,0 +1,441 @@
+<?php
+/**
+ * Annotations API: WP_Annotation_Caps class
+ *
+ * @package gutenberg
+ * @since [version]
+ */
+
+/**
+ * Annotation capabilities.
+ *
+ * @since [version]
+ */
+final class WP_Annotation_Caps {
```

Comment:
Great observation, thank you. I was thinking the same thing when I wrote this and tried to avoid some of that with the `get_comment_info()` and `get_post_info()` utils. But the extraction of those variables are now in duplicate and I'd like to improve. Really want to avoid the use of `extract()`. I'll dig in and see if there's anything I can do.

Reply:
I did some work on this in recent commits. Thanks!